### DISCUSSION TITLE
Consider: adapting smart filter to boost value of highly rated games

### DISCUSSION THREAD

**Discussion by sgstair** [ludumdare/ludumdare#2016] [General]
The argument is, we should get more visibility on games that are performing very well before voting closes, so we can get better statistical certainty on  the true value for their ratings.

One concern is, you could potentially use this change to infer something about your game's ratings before the voting period is over. One possible implementation to work around this issue would be to issue a fixed smart rating boost for games with any rating category value above some fixed level.

**Comment by sgstair** [comment]
(This was suggested by cerno-b in the thread at https://ldjam.com/events/ludum-dare/39/false-game/results-for-ludum-dare-39)

**Comment by mikekasprzak** [comment]
I had this same thought some time ago, but I didn't have a great concept for it.

The main issue is that scores are one of the magic data's. Calculating them isn't a cheap operation (yet). That's why other magic stuff only does 500 at a time, so we're not bogging things down. Also anything we include in magic is potentially public.

If a factor is included in the smart filter, given every other piece of public data, it's likely folks will be able to detect which games are being bumped by this. 

I kinda think this is something that should maybe be revisited later, once we've done some things that improve how active voters are. I don't have stat's but I'm certain that voting rates reeeeealy slow down after the first week. So switch some mode on like this might not be very effective, unless more people are actively voting during this time.

**Comment by Cerno-b** [comment]
I thought about this a bit more. One solution would be to compute the smart filter order every minute and store the results in the database. Then you would only have to sort by order.

A way to prevent people from guessing they are boosted because of high ratings would be to randomize the smart filter. Instead of treating the smart score as a fixed value to be sorted it would be used as a weight for randomization. You throw the dice every 5 minutes and games with 2 times the score of other games will be twice as likely to receive a high smart score. Not sure whether it's okay to keep the list static for so long.

A less intrusive way would be to leave everything as it is and pick a game randomly that is eligible for a boost. That game would then get a bonus on smart score that decreases over time. It will show up on the smart filter page 1 or 2 for about 10 minutes and then slowly drop down over time. There would be boost candidates sliding through the smart page for a short time which would limit people from realising a specific game being pushed.

It is probably a bit fiddly to set up but finding a covert way to give high rating/low vote games a boost in votes will get rid of a lot of animosity when the results are revealed

**Comment by Cerno-b** [comment]
Maybe the boost drop off is a bit too much effort though. A simple "show a random eligible game on the smart filter list page 2 for 15 minutes" may be easier to set up.

**Comment by sgstair** [comment]
I'm planning to add functionality to the API to get a list of random games, weighted by the smart filter, for your user to rate (so may exclude games you've already rated) - as a convenience function... 

We might be able to hide it by not adding it to the smart filter, but boosting it in the hidden weighted random function of this API, if this feature were to be used heavily.

**Comment by Cerno-b** [comment]
Sounds good.  

I assume that this view will only be available when logged in so it will also hide your own game which makes it even less obvious that you are boosted.

I guess there'll be a lot of trial and error involved in this one. Looking forward to seeing it in action.

**Comment by sgstair** [comment]
I've been planning to implement this rating boost in the weighted random game list API I implemented (above)
My current thinking on the direction for this is approximately as follows:
We'll compute weights for every game in the event. The weights will be something like:
`weight = smart_rating - compensation_factor + random_noise + boost` - All games above a certain weight will be included in the weighted random selection.
The boost will be a function of the game's current performance, and number of ratings, the intent is to boost games that have a high relative rating in one or more categories, and have not made it to a goal threshold (30 ratings, ideally)
The compensation factor and the scale of random noise and boost values can be tuned to adjust just how much more attention higher smart-rated games should get. Just using the raw smart rating may not prioritize the higher smart-rating games sufficiently. I'll try to clearly articulate the difference in effective view count vs difference in smart-rating and how the boost will affect this, and provide variables for tuning this, once the work is completed.

Another core goal of this work is to ensure that it's difficult to precisely infer what kinds of boosts are being applied. This will be implemented using rate limiting on the random game selection API, and I'll run some numbers to see how many samples will be necessary to discover the boosted games with any significant accuracy, to inform what the rate limits and random noise values will be set at.
